libs/core/tests/unit_tests/test_tools.py

```diff
@@ -2174,3 +2175,124 @@ def foo(x: int) -> Bar:
     assert foo.invoke(
         {"type": "tool_call", "args": {"x": 0}, "name": "foo", "id": "bar"}
     ) == Bar(x=0)
+
+
+def test_method_tool_self_ref() -> None:
+    """Test that a method tool can reference self."""
+
+    class A:
+        def __init__(self, c: int):
+            self.c = c
+
+        @MethodTool
+        def foo(self, a: int, b: int) -> int:
+            """Add two numbers to c."""
+            return a + b + self.c
+
+    a = A(10)
+    assert a.foo.invoke({"a": 1, "b": 2}) == 13
```

Comment:
good - let's also do some asserts on name/description/args schema

Reply:
Will do

Reply:
Have added one for this, let me know if it was exhaustive enough

Reply:
added as a code suggestion to be clear what this was suggesting - would be great to get these asserted across all cases!

Reply:
Thank you for the clarification, will get that done.